src/humancellatlas/data/metadata/helpers/staging_area.py

```diff
@@ -0,0 +1,330 @@
+                    'size': len(json.dumps(json_content)),
+                    'uuid': metadata_file.uuid,
+                    'version': metadata_file.version
+                }
+                manifest.append(file_manifest)
+                if entity_type.endswith('_file'):
+                    file_manifest = self.descriptors[entity_id].to_manifest_entry
+                    manifest.append(file_manifest)
+                else:
+                    pass
+        logger.debug('Bundle contents: %s', list(metadata.keys()))
+        return links_file.version, manifest, metadata
+
+    def _entity_ids_by_type(self,
+                            subgraph_id: str
+                            ) -> Mapping[str, Set[str]]:
+        """
+        Parse the links in a subgraph and return a mapping of the entity types
+        (e.g. 'analysis_file', 'cell_suspension') to a set of entity IDs
+        """
+        links_file: LinksFile = self.links[subgraph_id]
+        links_json = links_file.content
+        entity_ids = defaultdict(set)
+        # Project ID is only mentioned in the links JSON if there is a
+        # supplementary_file_link so add it in here to make sure it is included.
+        entity_ids['project'].add(links_file.project_id)
+        link: JSON
+        for link in links_json['links']:
+            link_type = link['link_type']
+            if link_type == 'process_link':
+                entity_type = link['process_type']
+                entity_id = link['process_id']
+                entity_ids[entity_type].add(entity_id)
+                for category in ('input', 'output', 'protocol'):
+                    for file in link[f'{category}s']:
+                        entity_type = file[f'{category}_type']
+                        entity_id = file[f'{category}_id']
+                        entity_ids[entity_type].add(entity_id)
+            elif link_type == 'supplementary_file_link':
+                for file in link['files']:
+                    entity_type = file['file_type']
+                    entity_id = file['file_id']
+                    entity_ids[entity_type].add(entity_id)
+            else:
+                raise ValueError('Unknown link type', link_type)
+        return dict(entity_ids)
+
+
+@attr.s(frozen=True, kw_only=True, auto_attribs=True)
+class GitHubStagingAreaFactory:
+    repo: Repository
+    ref: str
+    path: Tuple[str, ...]
+
+    @classmethod
+    def create(cls,
+               owner: str,
+               name: str,
+               ref: str,
+               path: Optional[str] = None
+               ) -> 'GitHubStagingAreaFactory':
+        """
+        :param owner: The owner of the GitHub repository
+        :param name: The name of the GitHub repository
+        :param ref: A branch name, tag, or commit SHA
+        :param path: The path inside the repository to the base of the staging area
+        """
+        token = os.environ['GH_API_TOKEN']  # A GitHub personal access token
```

Comment:
```suggestion
        token = os.environ['GITHUB_TOKEN']  # A GitHub access token
```

Sorry for the last minute renaming of the env var but this is in anticipation of switching to Github actions where the token is stored in a secret of the same name and I think it makes sense to use that name for the env var, too.